platform-sdk/swirlds-common/src/test/java/com/swirlds/common/wiring/component/WiringComponentPerformanceTests.java

```diff
@@ -0,0 +1,128 @@
+import com.swirlds.common.wiring.wires.input.InputWire;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.ForkJoinPool;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+@Disabled // Do not merge with this class enabled
+class WiringComponentPerformanceTests {
+
+    private interface SimpleComponent {
+        void handleInput(@NonNull Long input);
+    }
+
+    private static class SimpleComponentImpl implements SimpleComponent {
+        private long runningValue = 0;
+
+        @Override
+        public void handleInput(@NonNull final Long input) {
+            runningValue += input;
+        }
+
+        public long getRunningValue() {
+            return runningValue;
+        }
+    }
+
+    @NonNull
+    private InputWire<Long> buildOldStyleComponent(@NonNull final SimpleComponent component) {
+        final WiringModel model = WiringModel.create(
+                TestPlatformContextBuilder.create().build(), Time.getCurrent(), ForkJoinPool.commonPool());
+
+        final TaskScheduler scheduler = model.schedulerBuilder("test")
+                .withType(TaskSchedulerType.DIRECT)
+                .build();
+
+        final BindableInputWire<Long, Void> inputWire = scheduler.buildInputWire("input");
+        inputWire.bind(component::handleInput);
+
+        return inputWire;
+    }
+
+    @NonNull
+    private InputWire<Long> buildAutomaticComponent(@NonNull final SimpleComponent component) {
+
+        final WiringModel model = WiringModel.create(
+                TestPlatformContextBuilder.create().build(), Time.getCurrent(), ForkJoinPool.commonPool());
+
+        final TaskScheduler<Void> scheduler = model.schedulerBuilder("test")
+                .withType(TaskSchedulerType.DIRECT)
+                .build()
+                .cast();
+
+        final ComponentWiring<SimpleComponent, Void> componentWiring =
+                new ComponentWiring<>(SimpleComponent.class, scheduler);
+        final InputWire<Long> inputWire = componentWiring.getInputWire(SimpleComponent::handleInput);
+        componentWiring.bind(component);
+
+        return inputWire;
+    }
+
+    @Test
+    void oldStylePerformanceTest() {
+        final long iterations = 100_000_000;
+
+        final SimpleComponentImpl component = new SimpleComponentImpl();
+        final InputWire<Long> inputWire = buildOldStyleComponent(component);
+
+        final Instant start = Instant.now();
+
+        for (long i = 0; i < iterations; i++) {
+            inputWire.put(i);
+        }
+
+        final Instant end = Instant.now();
+        final Duration duration = Duration.between(start, end);
+        System.out.println("Time required: " + duration.toMillis() + "ms");
+
+        // Just in case the compiler wants to get cheeky and avoid doing computation
+        System.out.println("value = " + component.getRunningValue());
```

Comment:
can you add a comment with the current performance characteristics, so that we have something to compare to in the future

Reply:
comment added